modules/gatsby/src/gatsby-config/get-gatsby-config.js

```diff
@@ -323,6 +324,19 @@ module.exports = function getGatsbyConfig(config) {
 
   // conditional plug-ins - only added depending on options on config
 
+  if (paddedConfig.DOC_FOLDERS && paddedConfig.DOC_FOLDERS.length > 0) {
+    // Generates gatsby nodes for markdown files and JSON file in the in the docs folder
+    paddedConfig.DOC_FOLDERS.forEach(folderPath => {
+      gatsbyConfig.plugins.push({
+        resolve: 'gatsby-source-filesystem',
+        options: {
+          name: 'docs',
```

Comment:
what does this do?

Reply:
It will source files in the folder into Gatsby system and create `File` nodes. Later on, `gatsby-transformer-remark` transforms markdown files into `MarkdownRemark` nodes.

Here I'm trying to source data from `DOC_FOLDER` or `DOC_FOLDERS`

Reply:
Should these all have the same `name: 'docs'`? Does this affect how we query them?